server/data/redisClient.test.ts

```diff
@@ -0,0 +1,224 @@
+      }
+
+      mockRedisClient.json.get.mockResolvedValue(mockRedisData)
+
+      const result = await redisService.readLatest('component:latest')
+
+      expect(mockRedisClient.json.get).toHaveBeenCalledWith(
+        expect.objectContaining({ isolated: true }),
+        'component:latest',
+      )
+      expect(result).toEqual({
+        'some-service:dev': { v: '2023-02-02.123.abc1234', dateAdded: '2023-02-11' },
+        'some-service:prod': { v: '2023-01-15.120.abc1230', dateAdded: '2023-02-11' },
+      })
+    })
+
+    it('should handle empty Redis response', async () => {
+      mockRedisClient.json.get.mockResolvedValue({})
+
+      const result = await redisService.readLatest('empty-key')
+
+      expect(result).toEqual({})
+    })
+
+    it('should handle Redis read errors', async () => {
+      const error = new Error('Redis connection timeout')
+      mockRedisClient.json.get.mockRejectedValue(error)
+
+      await expect(redisService.readLatest('failing-key')).rejects.toThrow('Redis connection timeout')
+
+      expect(mockLogger.error).toHaveBeenCalledWith('Failed to read latest: Redis connection timeout', error)
+    })
+  })
+
+  describe('getAllDependencies', () => {
+    it('should successfully retrieve dependency information', async () => {
+      const mockDependencyInfo = {
+        components: {
+          'service-a': {
+            dependencies: ['service-b', 'database-x'],
+            dependents: ['service-c'],
+          },
+          'service-b': {
+            dependencies: ['database-y'],
+            dependents: ['service-a'],
+          },
+        },
+      }
+
+      mockRedisClient.json.get.mockResolvedValue(mockDependencyInfo)
+
+      const result = await redisService.getAllDependencies()
+
+      expect(mockRedisClient.json.get).toHaveBeenCalledWith(
+        expect.objectContaining({ isolated: true }),
+        'dependency:info',
+      )
+      expect(result).toBeDefined()
+      // Dependencies class constructor is called with the mock data
+    })
+
+    it('should handle dependency retrieval errors', async () => {
+      const error = new Error('Failed to fetch dependencies')
+      mockRedisClient.json.get.mockRejectedValue(error)
+
+      await expect(redisService.getAllDependencies()).rejects.toThrow('Failed to fetch dependencies')
+
+      expect(mockLogger.error).toHaveBeenCalledWith(
+        'Failed to get dependency info: Failed to fetch dependencies',
+        error,
+      )
+    })
+  })
+
+  describe('handleError', () => {
+    it('should handle ClientClosedError specifically', async () => {
+      const clientClosedError = new ClientClosedError()
+
+      await redisService.handleError(clientClosedError, 'Test operation failed')
+
+      expect(mockLogger.error).toHaveBeenCalledWith('The client is closed ...RECONNECTING')
+      expect(mockLogger.error).toHaveBeenCalledWith('Test operation failed: The client is closed', clientClosedError)
```

Comment:
we wouldn't usually test logging as it's too prone to changes so leads to sensitive tests